### PR TITLE
Integration tests for unschedulable nodes.

### LINKED ISSUE #6898: Create a test to verify that nothing will be scheduled on "Unschedulable" nodes
This was broken by #6717 and then (hopefully) fixed by #6857.
But we need a test for it.

cc @lavalamp @davidopp

### PR DESCRIPTION
**Pull request by wojtek-t**
Fix #6898

### CODE REVIEW

**Comment by wojtek-t** [comment]
I'm not sure that's the best way to test it - please let me know if you come up with a better idea.

**Comment by wojtek-t** [comment]
It seems that this test doesn't really work, but I will wait with fixing it for some possible suggestions about the approach from you.

**Comment by posita** [comment]
@wojtek-t, I'm not sure Shippable or Travis are working right now. I'm seeing a high failure rate in Shippable and a lower but significant failure rate in Travis on almost all PRs with pushes from the past 18-ish hours (including [my own](/GoogleCloudPlatform/kubernetes/pull/6879)).

**Comment by wojtek-t** [comment]
@posita I agree that especially shippable is really flaky now. But in case of verify-gofmt we can trust them.

**Comment by lavalamp** [comment]
Actually I was kinda hoping to add this to the test/integration/* framework-- is it easy to adapt to that? it's a bit nicer than this setup. Especially since you don't need any actual kubelets to test this, just apiserver & scheduler. Sorry for not being clear earlier :(

**Comment by lavalamp** [comment]
I think what you're doing here will break the other integration tests. Here's what I recommend:

In test/integration, start a new file. You can see from the other files there how to make an apiserver, verify a test etcd instance is running, and you'll need to make a scheduler.

Then add some unschedulable nodes, try to add a pod to the system, and verify that it doesn't schedule.

**Comment by wojtek-t** [comment]
I don't think it will break other tests, because I explicitly moved so that it's not running simultanuously with other tests. But I agree that what you suggested is better (in fact I wasn't aware that we have some other integration tests under test/integration). Will take a look and update this PR.

**Comment by wojtek-t** [comment]
@lavalamp 
I've change the test so that it is part of test/integration.
However, I think I need some help. The test is "mostly" working, but due to some strange reasons I don't understand it hangs forever on shutdown. This is definitely connected with how scheduler is created (removing scheduler helps).
However, when I create the scheduler and the test itself doesn't send any requests to apiserver, this test finishes.
In other words, the test fails only if both conditions are satisfied
- the scheduler is created
- I'm trying to send any request to apiserver
Any hints what I'm doing wrong?

**Comment by wojtek-t** [comment]
To be clear - the fact that this test passed on Travis is a surprise for me - it never passed for me [it was always hanging on shutdown]

**Comment by wojtek-t** [comment]
Actually I don't understand it - why travis is green if this test failed there? [Looking at the logs, it really FAILED].

**Comment by lavalamp** [comment]
Wow, that is really bad-- not sure why that didn't register as a failure. I will look at this after lunch.

File: test/integration/scheduler_test.go [hunk 1]
```diff
@@ -0,0 +1,149 @@
...
+
+	schedulerConfigFactory := factory.NewConfigFactory(client)
+	schedulerConfig, err := schedulerConfigFactory.Create()
+	if err != nil {
+		t.Fatalf("Couldn't create scheduler config: %v", err)
+	}
+	eventBroadcaster := record.NewBroadcaster()
+	schedulerConfig.Recorder = eventBroadcaster.NewRecorder(api.EventSource{Component: "scheduler"})
+	eventBroadcaster.StartRecordingToSink(client.Events(""))
+	scheduler.New(schedulerConfig).Run()
```
Comment (lavalamp) [inline test/integration/scheduler_test.go]: Does a ```defer close(schedulerConfig.StopEverything)``` help?

File: test/integration/scheduler_test.go [hunk 2]
```diff
@@ -0,0 +1,149 @@
...
+		KubeletClient:     client.FakeKubeletClient{},
+		EnableLogsSupport: false,
+		EnableUISupport:   false,
+		EnableIndex:       true,
+		APIPrefix:         "/api",
+		Authorizer:        apiserver.NewAlwaysAllowAuthorizer(),
+		AdmissionControl:  admit.NewAlwaysAdmit(),
+	})
+
+	deleteAllEtcdKeys()
```
Comment (lavalamp) [inline test/integration/scheduler_test.go]: perhaps put at the very top?

Comment (wojtek-t) [inline test/integration/scheduler_test.go]: Done.

File: test/integration/scheduler_test.go (hunk 1 above)
Comment (wojtek-t) [inline test/integration/scheduler_test.go]: Yes - it helps. Thanks!

**Comment by wojtek-t** [comment]
Comments addresses (test is passing now).
PTAL

**Comment by lavalamp** [comment]
LGTM
